src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/PreloginSubHandlers/PreloginPacketHandler.cs

```diff
@@ -0,0 +1,412 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.Common;
+using Microsoft.Data.SqlClient;
+using Microsoft.Data.SqlClientX.IO;
+
+namespace Microsoft.Data.SqlClientX.Handlers.Connection.PreloginSubHandlers
+{
+    internal class PreloginPacketHandler : IHandler<PreLoginHandlerContext>
+    {
+        const int GUID_SIZE = 16;
+
+        const int PAYLOAD_OFFSET_AND_LENGTH_SIZE_IN_BYTES = 4;
+
+        const int PAYLOAD_LENGTH_SIZE_IN_BYTES = 2;
+
+        // EventSource counter
+        private static int s_objectTypeCount;
+
+        internal readonly int _objectID = Interlocked.Increment(ref s_objectTypeCount);
+
+        internal int ObjectID => _objectID;
```

Comment:
This is only used in the tracing of `ReadPreLoginResponse`, by which time the connection GUID has been generated. I think we can replace `ObjectID` with the connection GUID and remove the property. The `ObjectID` doesn't make as much sense with the handler-driven architecture anyway - the previous design ensured that every connection had the same ID.

Reply:
@edwardneal  Object Ids are ints which are easier to correlate, when we get dumps/traces from customer cases. Guids would work too, but its easier to say "check this object with Object Id 1234", than to say "Check this object with Guid <SomethingThatDoesn'tRollOffTheTongue> " 
Based on the prior experience of debugging traces for SqlClient, I am biased towards the usage of ObjectID. Guids will work too, but its easier with ObjectIDs

I think the Object IDs can be moved to a base handler class, so that each handler gets an object id. 

@cheenamalhotra  any inputs here?

Reply:
That's fair enough - in which case you probably need the object ID to uniquely identify the connection as a whole, perhaps setting a property on the `ConnectionHandlerContext`. At the moment the ID identifies a specific handler.

Reply:
I agree with moving Object Id to orchestrator/context where all handlers can consume from.